### PR TITLE
Depend on Polyline.framework

### LINKED ISSUE #11: Use Polyline dependency
Just like MapboxDirections.swift, we should depend on the Polyline lib

### PR DESCRIPTION
**Pull request by 1ec5**
Added a dependency on raphaelmor/Polyline@f86f34dac9aae7e027fd9f62c2e01dd77f1ee312. (We can pin to a newer version or release once raphaelmor/Polyline#40 lands.)

Fixes #11.

/cc @incanus

### CODE REVIEW

File: MapboxStaticTests/MapboxStaticTests.swift [hunk 1]
```diff
@@ -370,7 +370,7 @@ class MapboxStaticTests: XCTestCase {
         let fillColor = Color.red
         let fillColorRaw = "ff0000"
         let fillOpacity = 0.25
-        let encodedPolyline = "(upztG%60jxkVn@al@bo@nFWzuAaTcAyZen@)"
+        let encodedPolyline = "(upztG%60jxkVn@al@bo@pFWzuAaTcAyZgn@)"
```
Comment (1ec5) [inline MapboxStaticTests/MapboxStaticTests.swift]: @tmcw, should we be concerned that there were differences in how Polyline encoded the polyline versus our homegrown encoder’s version? In this case at least, there didn’t appear to be any visual difference.

**Comment by 1ec5** [comment]
@bsudekum, this PR will make it much easier to upgrade to Swift 4.

File: Cartfile [hunk 2]
```diff
@@ -0,0 +1,2 @@
+github "raphaelmor/Polyline" ~> 4.2
```
Comment (bsudekum) [inline Cartfile]: This version is explicitly for swift 4. I'm not sure this will pass unless this changed to v4.1.1.

Comment (1ec5) [reply Cartfile]: I just upgraded the Bitrise apps to use Xcode 9 instead of Xcode 8.3. The builds should pass now.

**Comment by 1ec5** [comment]
This test currently fails:

```
MapboxStaticTests.ClassicOverlayTests
  testGeoJSONObject, XCTAssertNotNil failed - 
  /Users/vagrant/git/MapboxStaticTests/ClassicOverlayTests.swift:108
        
        XCTAssertNotNil(Snapshot(options: options, accessToken: BogusToken).image)
    }
```

**Comment by 1ec5** [comment]
Upgrading to Xcode 9 means the macOS 10.13 SDK uses different RGB component values for all the NSColor constants than UIColor does. 🙄

**Comment by 1ec5** [comment]
Fixed the tests in 179016cd4ccfa94f8fe61b9d73de04d8a65a239d.

In the iOS 11 and macOS 10.13 SDKs, JSONSerialization adds an extra digit of precision. Also, in the macOS 10.13 SDK, the RGB components of NSColor’s class properties differ from those of UIColor’s similarly named class properties.

**Comment by 1ec5** [comment]
@bsudekum @frederoni @friedbunny, this PR is ready for review.

File: MapboxStatic.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1071,6 +1125,10 @@
 				DYLIB_COMPATIBILITY_VERSION = 1;
 				DYLIB_CURRENT_VERSION = 7;
 				DYLIB_INSTALL_NAME_BASE = "@rpath";
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(inherited)",
+					"$(PROJECT_DIR)/Carthage/Build/iOS",
```
Comment (frederoni) [inline MapboxStatic.xcodeproj/project.pbxproj]: This works but I think the recommended way is https://github.com/Carthage/Carthage#if-youre-building-for-ios-tvos-or-watchos

**Pull request merged**